### PR TITLE
rpc: set default bip32derivs to true for psbt methods

### PR DESCRIPTION
**Pull request by Sjors**
In https://github.com/bitcoin/bitcoin/pull/13557#pullrequestreview-135905054 I recommended not including bip32 deriviation by default in PSBTs:

> _Bit of a privacy issue_: let's say person A and B are about to spend from a multisig address, sending everything to person A. Person A gives their address to person B, their wallet wallet creates a PSBT, but doesn't sign it. Wallet A then calls `walletprocesspsbt` which signs it and _spontaneously adds the master_fingerprint and bip32 path_. Same issue with `walletcreatefundedpsbt`.
> 
> Adding `bip32_derivs` should probably be opt-in.

In practice I find this default quite annoying because I forget it and end up with a confused hardware wallet.

More importantly, in the multisig example I provided, it's actually essential for the other side to know the derivation details (in addition to an xpub). This allows them to check that change is going to an address you can still co-sign for (because the spending policy is unchanged except for an index).

### CODE REVIEW

**Comment by jb55** [comment]
Concept/Approach ACK 29a21c90610aed88b796a7a5900e42e9048b990e

**Comment by instagibbs** [comment]
imo the privacy ship has sailed on PSBTs, we have no functionality to *remove* data from PSBTs yet, and no conceptual workflow to manage such a thing. We have thought about this in context of Confidential Assets as well dealing with blinded amounts/assets. It's complicated.

Therefore, concept ACK

**Comment by sipa** [comment]
Privacy in PSBTs is hard; if done properly, it should probably mean having different domains of participants/signers, and filtering out private information when crossing boundaries. I don't think it's simple, or something that can be delegated to users using a flag at signing time (because usually you'd want to reveal that derivation information to e.g. your hardware wallet but not to other signers).

Concept ACK.

**Comment by achow101** [comment]
ACK 29a21c90610aed88b796a7a5900e42e9048b990e

**Comment by sipa** [comment]
Code review ACK 29a21c90610aed88b796a7a5900e42e9048b990e. The changes look straightforward, but a test that these fields are filled would be nice.

**Comment by Sjors** [comment]
I added a test to demonstrate the default behavior. Good call, because I discovered #18039 in the process.

**Comment by Sjors** [comment]
I won't rebase, to preserve ACKs, but I did run the test suite locally on a rebased version.

**Review by jonatack — COMMENTED** [review]
ACK 5bad7921d0 code review, built, ran tests, inspected/messed around with/pprinted values from the new tests. Thanks for adding the tests.

Edit: also built and ran tests after rebasing onto current master @ eb3c6b09

**Comment by jonatack** [comment]
I wonder if a deprecation process would be good here.

File: test/functional/rpc_psbt.py [hunk 1]
```diff
@@ -192,12 +192,20 @@ def run_test(self):
         psbt_orig = self.nodes[0].createpsbt([{"txid":txid1,  "vout":vout1}, {"txid":txid2, "vout":vout2}], {self.nodes[0].getnewaddress():25.999})
 
         # Update psbts, should only have data for one input and not the other
-        psbt1 = self.nodes[1].walletprocesspsbt(psbt_orig)['psbt']
+        psbt1 = self.nodes[1].walletprocesspsbt(psbt_orig, False, "ALL")['psbt']
```
Comment (instagibbs) [inline test/functional/rpc_psbt.py]: why is it important that signing is `False` and specified sighash flag??

**Review by instagibbs — COMMENTED** [review]
logic ACK, leaving a question about the test

File: test/functional/rpc_psbt.py (hunk 1 above)
Comment (Sjors) [reply test/functional/rpc_psbt.py]: I vaguely remember running into a problem specific to sign is `False`, but I can't remember.

Comment (Sjors) [reply test/functional/rpc_psbt.py]: Because otherwise the test runs into #18039

**Comment by instagibbs** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/17264/commits/5bad7921d0b33b62c0a59a478c2e8c869fc5e3b5

**Review by meshcollider — COMMENTED** [review]
utACK 5bad7921d0b33b62c0a59a478c2e8c869fc5e3b5

**Review by KaycoinsInc — APPROVED** [review]

**Pull request merged**